tests/test_labelarray.py

```diff
@@ -337,3 +339,37 @@ def test_setitem_array(self):
         # Write the whole array.
         arr[:] = orig_arr
         check_arrays(arr, orig_arr)
+
+    def test_narrow_code_storage(self):
+        def create_categories(width, plus_one):
+            length = int(width / 8) + plus_one
+            return [
+                ''.join(cs)
+                for cs in take(
+                    2 ** width + plus_one,
+                    product([chr(c) for c in range(256)], repeat=length),
+                )
+            ]
+
+        #  uint8
+        categories = create_categories(8, plus_one=False)
```

Comment:
Is this testing that we use a uint8 when we have exactly 8 unique categories?  I think that should only be true if the categories already includes the missing value, otherwise we'll overflow on https://github.com/quantopian/zipline/pull/1672/files#diff-feac3c74134e7ddfc5a86bb82804193cL21.